packages/eslint/src/generators/utils/eslint-file.ts

```diff
@@ -207,7 +207,7 @@ function overrideNeedsCompat(
   override: Linter.ConfigOverride<Linter.RulesRecord>
 ) {
   return (
-    !override.env && !override.extends && !override.plugins && !override.parser
+    override.env || override.extends || override.plugins || override.parser
```

Comment:
This seems unrelated?

Reply:
As mentioned, removing this temporary patch resurfaced some minor issues in eslint construction which was fixed as a part of this (otherwise the tests would fail).